includes/class-sensei-course-structure.php

```diff
@@ -300,13 +304,46 @@ public function save( array $raw_structure ) {
 	 */
 	private function save_module( array $item ) {
 		if ( $item['id'] ) {
-			$term = get_term( $item['id'], 'module' );
-			$slug = $this->get_module_slug( $item['title'] );
+			$term         = get_term( $item['id'], 'module' );
+			$author       = get_post( $this->course_id )->post_author;
+			$default_slug = $this->get_module_slug( $item['title'] );
+			// Custom slug gets priority over conventional slug.
+			$slug = $item['slug'] ?? $default_slug;
 
-			// Slug has changed.
 			if ( $term->slug !== $slug ) {
-				$existing_module_id = $this->get_existing_module( $item['title'] );
-				$module_id          = $existing_module_id ? $existing_module_id : $this->create_module( $item );
+				// First try to get an existing module directly by the provided slug.
+				$existing_module = get_term_by( 'slug', $slug, 'module' );
+				// If not found, try to find if this teacher already has a module for this, with the default slug.
+				if ( ! $existing_module && $slug !== $default_slug ) {
+					$existing_module = get_term_by( 'slug', $default_slug, 'module' );
+				}
+				// If not found, try to find module for this teacher using term-meta.
+				if ( ! $existing_module && ! user_can( $author, 'manage_options' ) ) {
+					$modules = get_terms(
+						array(
+							'taxonomy'   => 'module',
+							'name'       => $item['title'],
+							'hide_empty' => false,
+							'meta_query' => array( // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query -- Only executed while saving, not too many data possibly.
+								array(
+									'key'     => 'module_author',
+									'value'   => $author,
+									'compare' => '=',
+								),
+							),
+						)
+					);
+					if ( ! empty( $modules ) ) {
+						$existing_module = $modules[0];
+					}
```

Comment:
What do you think about extracting the "get existing module" logic to a separate method?